src/main/java/com/openshift/jenkins/plugins/pipeline/OpenShiftBuilder.java

```diff
@@ -20,165 +21,192 @@
-         */
-        public FormValidation doCheckApiURL(@QueryParameter String value)
-                throws IOException, ServletException {
-        	return ParamVerify.doCheckApiURL(value);
-        }
-
-        public FormValidation doCheckBldCfg(@QueryParameter String value)
-                throws IOException, ServletException {
-        	return ParamVerify.doCheckBldCfg(value);
-        }
-
-        public FormValidation doCheckNamespace(@QueryParameter String value)
-                throws IOException, ServletException {
-        	return ParamVerify.doCheckNamespace(value);
-        }
-        
-        public FormValidation doCheckCheckForTriggeredDeployments(@QueryParameter String value)
-                throws IOException, ServletException {
-        	return ParamVerify.doCheckCheckForTriggeredDeployments(value);
-        }
-        
-        public FormValidation doCheckWaitTime(@QueryParameter String value)
-                throws IOException, ServletException {
-        	return ParamVerify.doCheckCheckForWaitTime(value);
-        }
-        
-        public FormValidation doCheckAuthToken(@QueryParameter String value)
-                throws IOException, ServletException {
-        	return ParamVerify.doCheckToken(value);
-        }
-
-        public boolean isApplicable(Class<? extends AbstractProject> aClass) {
-            // Indicates that this builder can be used with all kinds of project types 
-            return true;
-        }
-
-        /**
-         * This human readable name is used in the configuration screen.
-         */
-        public String getDisplayName() {
-            return DISPLAY_NAME;
-        }
-        
-        public long getWait() {
-        	return wait;
-        }
-
-        @Override
-        public boolean configure(StaplerRequest req, JSONObject formData) throws FormException {
-            // To persist global configuration information,
-            // pull info from formData, set appropriate instance field (which should have a getter), and call save().
-        	wait = formData.getLong("wait");
-        	GlobalConfig.setBuildWait(wait);
-            save();
-            return super.configure(req,formData);
-        }
-
-    }
+	// If your plugin doesn't really define any property on Descriptor,
+	// you don't have to do this.
+	@Override
+	public DescriptorImplIOpenShift getDescriptor() {
+		return (DescriptorImplIOpenShift)super.getDescriptor();
+	}
+
+	/**
+	 * Descriptor for {@link OpenShiftBuilder}. Used as a singleton.
+	 * The class is marked as public so that it can be accessed from views.
+	 *
+	 */
+	// NOTE: The getters in this class are used to populate the values of global.jelly, and the fields must match the
+	// method names
+	@Extension // This indicates to Jenkins that this is an implementation of an extension point.
+	public static final class DescriptorImplIOpenShift extends BuildStepDescriptor<Builder> implements IOpenShiftTimedStepDescriptor {
```

Comment:
Interesting - I was always under the assumption that there was some naming restriction and that it had to be `DescriptorImpl`.  Good to know based on your testing that assumption is in fact incorrect.